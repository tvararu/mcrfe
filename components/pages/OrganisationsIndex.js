import Head from "next/head";
import gql from "graphql-tag";
import Query from "../elements/Query";
import Link from "next/link";
import GovukMain from "../govuk/Main";

const allProvidersQuery = gql`
  {
    allProviders(orderBy: PROVIDER_NAME_ASC) {
      nodes {
        id
        providerCode
        providerName
        coursesByProviderId {
          totalCount
        }
      }
    }
  }
`;

const OrganisationListItem = ({
  providerCode: code,
  providerName: name,
  coursesByProviderId: { totalCount: courseCount }
}) => (
  <li>
    <h2 className="govuk-heading-m">
      <Link
        as={`/organisations/${code}`}
        href={`/organisations?providerCode=${code}`}
        prefetch
      >
        <a className="govuk-link">{name}</a>
      </Link>
      <span className="govuk-body govuk-!-font-weight-regular govuk-!-display-block">
        {courseCount} {courseCount !== 1 ? "courses" : "course"}
      </span>
    </h2>
  </li>
);

const OrganisationsList = () => (
  <Query query={allProvidersQuery}>
    {({ data }) => (
      <ul className="govuk-list">
        {data.allProviders.nodes.map(provider => (
          <OrganisationListItem key={provider.id} {...provider} />
        ))}
      </ul>
    )}
  </Query>
);

export default () => (
  <GovukMain>
    <Head>
      <title>Organisations - Publish teacher training courses - GOV.UK</title>
    </Head>
    <h1 className="govuk-heading-xl">Organisations</h1>
    <OrganisationsList />
  </GovukMain>
);
