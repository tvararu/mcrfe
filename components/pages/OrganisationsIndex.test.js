import renderer from "react-test-renderer";
import OrganisationsIndex from "./OrganisationsIndex";

jest.mock("../elements/Query", () => ({ children }) =>
  children({
    data: {
      allProviders: {
        nodes: [
          {
            id: 1,
            providerCode: "A01",
            providerName: "Acme",
            coursesByProviderId: { totalCount: 0 }
          },
          {
            id: 2,
            providerCode: "A02",
            providerName: "Bat",
            coursesByProviderId: { totalCount: 1 }
          }
        ]
      }
    }
  })
);

describe("OrganisationsIndex", () => {
  it("renders correctly", () => {
    const component = renderer.create(<OrganisationsIndex />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
