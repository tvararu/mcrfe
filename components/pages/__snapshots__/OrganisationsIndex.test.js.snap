// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationsIndex renders correctly 1`] = `
<main
  className="govuk-main-wrapper"
  id="main-content"
  role="main"
>
  <h1
    className="govuk-heading-xl"
  >
    Organisations
  </h1>
  <ul
    className="govuk-list"
  >
    <li>
      <h2
        className="govuk-heading-m"
      >
        <a
          className="govuk-link"
        >
          Acme
        </a>
        <span
          className="govuk-body govuk-!-font-weight-regular govuk-!-display-block"
        >
          0
           
          courses
        </span>
      </h2>
    </li>
    <li>
      <h2
        className="govuk-heading-m"
      >
        <a
          className="govuk-link"
        >
          Bat
        </a>
        <span
          className="govuk-body govuk-!-font-weight-regular govuk-!-display-block"
        >
          1
           
          course
        </span>
      </h2>
    </li>
  </ul>
</main>
`;
